ion/src/main/java/com/fasterxml/jackson/dataformat/ion/IonParser.java

```diff
@@ -331,31 +338,55 @@ public int getTextOffset() throws IOException {
 
     @Override
     public BigInteger getBigIntegerValue() throws IOException {
+        NumberType nt = getNumberType();
```

Comment:
I think I'd like handling of this part to be bit different: could you have a separate PR for `getXXXValue()` changes, so I can fix other NPE parts from above?
(I can suggest alternative way on that PR without slowing merging of this one)

Reply:
Actually never mind; simple enough to isolate check, updated PR and will merge. Thanks again!

Reply:
Thanks.

Reply:
Thank you @arthurscchan for figuring out the root cause & providing fix!